clients/central/xrt.go

```diff
@@ -99,12 +100,13 @@ func NewCommandOptions(commandTopic string) *CommandOptions {
 	}
 }
 
-func NewDiscoveryOptions(discoveryTopic string, discoveryMessageHandler MessageHandler, discoveryDuration, discoveryTimeout time.Duration) *DiscoveryOptions {
+func NewDiscoveryOptions(discoveryTopic string, discoveryMessageHandler MessageHandler, discoveryDuration, discoveryTimeout time.Duration, extendedDiscoveryOptions map[string]any) *DiscoveryOptions {
```

Comment:
To reduce the breaking change, create another function:
`func NewDiscoveryOptionsExtended(discoveryTopic string, discoveryMessageHandler MessageHandler, discoveryDuration, discoveryTimeout time.Duration, extendedDiscoveryOptions map[string]any) *DiscoveryOptions`

For the old one, pass an empty map to `ExtentedDiscoveryOptions`
`ExtentedDiscoveryOptions: make(map[string]any),`